src/System.Linq.Expressions/src/System/Dynamic/Utils/TypeExtensions.cs

```diff
@@ -44,6 +37,11 @@ private static bool MatchesArgumentTypes(this MethodInfo mi, Type[] argTypes)
             Debug.Assert(mi != null);
```

Comment:
Because debug assertion failures take the whole thing down, it doesn't get to write the results file the way it would if a test assertion failure.